src/views/List.jsx

```diff
@@ -1,15 +1,60 @@
 import { ListItem } from '../components';
+import { useState } from 'react';
+import { useEffect } from 'react';
 
 export function List({ data }) {
+	const [searchQuery, setSearchQuery] = useState('');
+	const [searchResults, setSearchResults] = useState([]);
+	// user type in search item - query
+	// as letters come in, filters items rendered on page
+	// set query to searchQuery
+	// render onto page search Results -> setting whatever is filtered into searchResults
+
+	function filterResults(query) {
+		return data.filter((item) => item.name.toLowerCase().includes(query));
+	}
+
+	useEffect(() => {
+		setSearchResults(filterResults(searchQuery));
+
+		// ignoring dependency array warning for now
+		// adding filterResults causes infinite re-render
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [searchQuery]);
```

Comment:
Nice use of this useEffect! I bet that was tricky to figure out!

Reply:
I have a question about what does [searchQuery] in useEffect do? Asking to get a better understanding of how this hook works.

Reply:
It's in the dependency array - this means that every time searchQuery is changed, the useEffect is triggered.

Reply:
Yep, what Mira said. Any variable you put in the array will basically tell the useEffect to trigger any time that variable changes. Therefore, in this case.. we want our setSearchResults to be triggered every time the searchQuery changes i.e. on each new key press. If I type 'a', then 'p', then 'p', the search results will have been set for a search query of 'a', then 'ap', then 'app', so on and so forth. This would include backspaces, as well as clearing the string with our Clear Search button. 

Here is an article that seems to explain dependency arrays pretty well:
https://devtrium.com/posts/dependency-arrays

Reply:
thank you for the explanation!